core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -4644,19 +4644,21 @@ public RubyString pack(ThreadContext context, IRubyObject obj) {
     public RubyString pack(ThreadContext context, IRubyObject obj, IRubyObject maybeOpts) {
         Ruby runtime = context.runtime;
         IRubyObject opts = ArgsUtil.getOptionsArg(runtime, maybeOpts);
-        RubyString str = RubyString.newEmptyString(runtime);
-        RubyString iFmt = obj.convertToString();
+        RubyString str;
+
         try {
-            if (!opts.isNil()){
-                IRubyObject buffer = ((RubyHash) opts).fastARef(runtime.newSymbol("buffer"));
-                if (buffer != null) {
-                    str = (RubyString) buffer;
-                }
+            IRubyObject buffer = ((RubyHash) opts).fastARef(runtime.newSymbol("buffer"));
+            if (buffer != null) {
+                str = (RubyString) buffer;
+            } else {
+                str = RubyString.newString(runtime, "");
             }
+            return str.append(pack(context, obj));
 
-            return str.append(Pack.pack(context, context.runtime, this, iFmt));
         } catch (ArrayIndexOutOfBoundsException e) {
             throw concurrentModification(context.runtime, e);
+        } catch (ClassCastException e){
```

Comment:
In my original implementation I hadn't considered what would happen if the buffer were nil or some other class besides String.  MRI Ruby throws a type error, and this seemed to be the most straightforward way of mirroring that behavior.